Sources/Tracking/migration/DataPipelineMigrationAssistant.swift

```diff
@@ -20,12 +27,34 @@ class DataPipelineMigrationAssistant {
         logger: Logger,
         queue: Queue,
         jsonAdapter: JsonAdapter,
-        threadUtil: ThreadUtil
+        threadUtil: ThreadUtil,
+        profileStore: ProfileStore
     ) {
         self.logger = logger
         self.backgroundQueue = queue
         self.jsonAdapter = jsonAdapter
         self.threadUtil = threadUtil
+        self.profileStore = profileStore
+    }
+
+    func handleMigration() {
+        handleAlreadyIdentifiedMigratedUser()
+        handleQueueBacklog()
+    }
+
+    func handleAlreadyIdentifiedMigratedUser() {
```

Comment:
Besides migrating the profile ID, does our SDK also need to migrate the currently registered device token? 

If so, it can be in a separate PR.

Reply:
Device token should be updated on [initialising the SDK](https://github.com/customerio/customerio-ios/blob/10168a3e6807c52404d66e96aeb49ec881f3956d/Sources/Common/CustomerIOInstance.swift#L128) from stored value in globalDataStore.

Reply:
If this make sense to you then please feel free to resolve the conversation.